### ISSUE TITLE
Helix deployment script no longer working due to security changes

### ISSUE THREAD

**Issue by v-parose** [dotnet/dnceng#3622]
### Release Note Category
- [x] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements
### Release Note Description

Due to the recent security changes the way the Helix agent is deployed is no longer working. The DDFUN team needs to be updated on the new process the .NET team is developing.

Old Helix deployment script method was to run the below on the machines:

Windows machines:
powershell Invoke-WebRequest https://helixosauto.blob.core.windows.net/bootstrap/run.bat -OutFile .\run.bat && .\run.bat --production --skip-platform-check -q (HELIX QUEUE) --id (APP ID) 

Unix machines:
sudo curl https://helixosauto.blob.core.windows.net/bootstrap/run.sh -o ./run.sh && sudo chmod u+x ./run.sh && sudo ./run.sh -q (HELIX QUEUE) -p --id (APP ID)

The Application that was previously used has been deleted due to required security changes, so this will error when running it to deploy machines into Helix.

**Comment by ilyas1974** [comment]
@chcosta @missymessa @davfost can you please take a look at this? On Prem deployments are currently blocked due to app secret changes.

**Comment by missymessa** [comment]
None of the changes I've made have been merged into main, let alone deployed, and I've not deleted or changed any of the existing secrets.

**Comment by chcosta** [comment]
Messaged @v-parose to get a few more details.  I haven't made any changes in this space, but i'm happy to poke around a bit.

FYI @dougbu

**Comment by chcosta** [comment]
following up with DDFun

**Comment by ilyas1974** [comment]
This issue is that the script we provided DDFUN to set up machines does not meet the requirements for ID 1.1 - there is no specific change that caused this, but as part of the ID 1.1 work, we need to come up with a solution for DDFUN that works with the new requirements.

**Comment by riarenas** [comment]
> there is no specific change that caused this
If we had an application with a client secret, and we no longer have an application with a client secret, something definitely changed. I would definitely like to understand how an application that was still in use was deleted.

That doesn't change the fact we need a path forward, but this seems to have been a forced move rather than something planned.

**Comment by riarenas** [comment]
It looks like this requires a rewrite of the scripts in this directory: https://dnceng.visualstudio.com/internal/_git/dotnet-helix-machines?path=/src/PrepareOnPrem/scripts/bootstrap so that they take something other than an app id + secret as the auth mechanism. This is a non-trivial change and I will start looking at what is required.

**Comment by riarenas** [comment]
Current status:

The work to support a different authentication mechanism inside of the Helix on-prem scripts is non-trivial and is squarely SFI work, so we asked DDFUN to bring back the application they deleted, along with the creation of a new secret to unblock this scenario.

After working through the remaining issues (https://github.com/dotnet/dnceng/issues/3622 and some cleanup of the test machine I was given to investigate) I was able to run the scripts to completion and saw the machine start to take work in Helix.

This is ready to be attempted on new, clean machines.

**Comment by riarenas** [comment]
I've received confirmation that this is unblocked for now. I'll find where we are tracking the transition of this funcionality to a SFI friendly method and then close this out.
